src/ImageSharp/Numerics/RectangleF.cs

```diff
@@ -37,221 +33,291 @@ public struct RectangleF : IEquatable<RectangleF>
             set
             {
-                this.backingVector.Y = value;
+                this.Width = value.Width;
+                this.Height = value.Height;
             }
         }
 
         /// <summary>
-        /// Gets or sets the width of this <see cref="RectangleF"/>.
+        /// Gets a value indicating whether this <see cref="RectangleF"/> is empty.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public bool IsEmpty => (this.Width <= 0) || (this.Height <= 0);
+
+        /// <summary>
+        /// Gets the y-coordinate of the top edge of this <see cref="RectangleF"/>.
         /// </summary>
-        public float Width
+        public float Top
         {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
             get
             {
-                return this.backingVector.Z;
-            }
-
-            set
-            {
-                this.backingVector.Z = value;
+                return this.Y;
             }
         }
 
         /// <summary>
-        /// Gets or sets the height of this <see cref="RectangleF"/>.
+        /// Gets the x-coordinate of the right edge of this <see cref="RectangleF"/>.
         /// </summary>
-        public float Height
+        public float Right
         {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
             get
             {
-                return this.backingVector.W;
+                return this.X + this.Width;
             }
+        }
 
-            set
+        /// <summary>
+        /// Gets the y-coordinate of the bottom edge of this <see cref="RectangleF"/>.
+        /// </summary>
+        public float Bottom
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get
             {
-                this.backingVector.W = value;
+                return this.Y + this.Height;
             }
         }
 
         /// <summary>
-        /// Gets a value indicating whether this <see cref="RectangleF"/> is empty.
+        /// Gets the x-coordinate of the left edge of this <see cref="RectangleF"/>.
         /// </summary>
-        [EditorBrowsable(EditorBrowsableState.Never)]
-        public bool IsEmpty => this.Equals(Empty);
+        public float Left
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get
+            {
+                return this.X;
+            }
+        }
 
         /// <summary>
-        /// Gets the y-coordinate of the top edge of this <see cref="RectangleF"/>.
+        /// Creates a <see cref="RectangleF"/> with the coordinates of the specified <see cref="Vector4"/>.
         /// </summary>
-        public float Top => this.Y;
+        /// <param name="vector">The rectangle</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static implicit operator RectangleF(Vector4 vector) => new RectangleF(vector.X, vector.Y, vector.Z, vector.W);
```

Comment:
Not convinced these should implicitly (or even explicitly) convert to/from `Vector4` is this conversion used anywhere? 

The main reason is its not certain what will be mapped where, should a vector represent `x`,`y`,`width`,`height` or should it represent `x`,`y`,`right`,`left` because it ambiguous I would drop it, we can add some extension methods if there's anything that needs this.